snapper/XmlFile.cc

```diff
@@ -122,23 +123,38 @@ namespace snapper
     {
 	list<const xmlNode*> ret;
 
+	if (node != NULL)
+	    node = node->children;
 	for (const xmlNode* cur_node = node; cur_node; cur_node = cur_node->next)
 	{
 	    if (cur_node->type == XML_ELEMENT_NODE &&
 		strcmp(name, (const char*) cur_node->name) == 0)
 	    {
-		if (cur_node->children)
-		    ret.push_back(cur_node->children);
+		ret.push_back(cur_node);
 	    }
 	}
 
 	return ret;
     }
 
+    bool
+    getValue(const xmlNode* node, string& value)
+    {
+	xmlChar* tmp = xmlNodeGetContent(node);
+	if (tmp == nullptr)
+	    value = "";
+	else {
+	    value = (const char *) tmp;
+	    xmlFree(tmp);
+	}
+	return true;
```

Comment:
This always return true. Please explain.

Reply:
You're right, that was inconsistent. Now returning false if there's an error